tests/cupyx_tests/scipy_tests/special_tests/test_statistics.py

```diff
@@ -141,3 +139,158 @@ def check_unary_lower_precision(self, name, xp, scp, dtype):
+    def test_linspace_pdtri(self, xp, scp, int_dtype, dtype):
+        import scipy.special  # NOQA
+
+        func = getattr(scp.special, 'pdtri')
+        k = xp.arange(1, 10, dtype=int_dtype)[:, xp.newaxis]
+        y = xp.linspace(0, 1, 20, dtype=dtype)[xp.newaxis, :]
+        return func(k, y)
+
+    @pytest.mark.parametrize(
+        'function, args, expected, rtol',
+        [('chdtr', (1, 0), 0.0, None),
+         ('chdtr', (0.7, cupy.inf), 1.0, None),
+         ('chdtr', (0.6, 3), 0.957890536704110, 1e-12),
+         ('chdtrc', (1, 0), 1.0, None),
+         ('chdtrc', (0.6, 3), 1 - 0.957890536704110, 1e-12),
+         ('chdtri', (1, 1), 0.0, None),
+         ('chdtri', (0.6, 1 - 0.957890536704110), 3, 1e-12),
+         ]
+    )
+    def test_scalar(self, function, args, expected, rtol):
+        self._test_scalar(function, args, expected, rtol)
+
+
+@testing.gpu
+@testing.with_requires('scipy')
+class TestThreeArgumentDistributions(_TestDistributionsBase):
+
+    @pytest.mark.parametrize('function', ['btdtr', 'btdtri', 'fdtr', 'fdtrc',
+                                          'fdtri', 'gdtr', 'gdtrc'])
+    @testing.for_float_dtypes()
+    @testing.numpy_cupy_allclose(atol=1e-5, rtol=1e-5, scipy_name='scp')
+    def test_linspace_broadcasted(self, xp, scp, dtype, function):
+        """Linspace with three arguments.
+
+        This method uses first two arguments with mostly non-negative values.
+        In some cases, the last argument is constrained to range [0, 1]
+        """
+
+        import scipy.special  # NOQA
+
+        func = getattr(scp.special, function)
+        # a and b should be positive
+        a = xp.linspace(-1, 21, 30, dtype=dtype)[:, xp.newaxis, xp.newaxis]
+        b = xp.linspace(-1, 21, 30, dtype=dtype)[xp.newaxis, :, xp.newaxis]
+        if function in ['fdtri', 'btdtr', 'btdtri']:
+            # x should be in [0, 1] so concentrate values around that
+            x = xp.linspace(-0.1, 1.3, 20, dtype=dtype)
+        else:
+            # x should be non-negative, but test with at least 1 negative value
+            x = xp.linspace(-1, 10, 20, dtype=dtype)
+        x = x[xp.newaxis, xp.newaxis, :]
+        return func(a, b, x)
+
+    @pytest.mark.parametrize('function', ['bdtr', 'bdtrc', 'bdtri'])
+    @testing.for_float_dtypes()
+    @testing.for_signed_dtypes(name='int_dtype')
+    @testing.numpy_cupy_allclose(atol=1e-5, rtol=1e-5, scipy_name='scp')
+    def test_binomdist_linspace(self, xp, scp, function, dtype, int_dtype):
+        import scipy.special  # NOQA
+
+        if xp.dtype(int_dtype) not in [xp.int32, xp.int64]:
+            if xp.dtype(dtype) != xp.float64:
+                # Skip cases deprecated in SciPy 1.7+ via this Cython code:
+                # https://github.com/scipy/scipy/blob/cdb9b034d46c7ba0cacf65a9b2848c5d49c286c4/scipy/special/_legacy.pxd#L39-L43  # NOQA
+                # It causes infinite recursion in numpy_cupy_allclose
```

Comment:
This seems to be not solved in windows ci with scipy 1.6 :)

Reply:
I did also find a change in behavior for SciPy < 1.5, so I added a `with_requires('scipy>=1.5.0')` on this test case. Given that the failure occurred for SciPy 1.6.3 that should be unrelated, though.

I did fix a difference in the order of type casting cases in the ufunc in 18f1436, so maybe worth trying again? Doesn't seem likely that this would be Windows-specific though.

Reply:
The actual issue may be related to `long` being 32-bit instead of 64-bit on that platform? That might cause a different type cast to occur there than on the linux system I have been testing with locally.

Reply:
I just verified on a Windows laptop that `np.dtype(int).itemsize` is 4 and not 8, so I think character code 'l' as in the ufunc type codes is also 32-bit there which is probably the source of the difference. I have attempted a fix in 7a3652a, but couldn't test it locally as there is no CUDA-capable GPU in that laptop.

Reply:
Thanks for the effort in reproducing this!!! 😃
I think in this case you can use `numpy.dtype('q')` that gives the same int64 type in linux and windows

Reply:
I look at the following to see what types SciPy has defined:
```Python
from scipy import special

print(special.bdtr.types)
```
which gives `['fff->f', 'dld->d', 'ddd->d']` which is where the `l` type is coming in. I assume that on Windows this means a 64-bit int will go to the `ddd->d` path instead `dld->d` since 'l' is 32-bit on that system?

Reply:
I think so, but shouldn't scipy ensure that `dld` is also executed in windows?